takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/compile/jdt/CompilerJdt.java

```diff
@@ -684,14 +684,13 @@ public boolean setSourcepath(List<File> dependencies, Set<File> sourceRoots) thr
   public boolean setProcessorpath(List<File> processorpath) throws IOException {
     if (processorpath == null) {
       this.processorpath = dependencies;
-    } else if (processorpath.isEmpty()) {
-      this.processorpath = Collections.emptyList();
     } else {
-      throw new IllegalArgumentException();
+      this.processorpath = ImmutableList.copyOf(processorpath);
     }
     if (!isProcNone() && processorpathDigester.digestProcessorpath(this.processorpath)) {
       log.debug("Annotation processor path changed, recompiling all sources");
       strategy.enqueueAllSources();
+      strategy = new FullCompilationStrategy();
```

Comment:
Agree this line does not belong to this change, will remove before merge.

The line is still useful because `FullCompilationStrategy` has much simpler logic, but I need to introduce "escalate to full compile" helper instead. In a future commit.